src/content/1.7/development/architecture/migration-guide/forms/SQRS-usage-in-forms.md

```diff
@@ -0,0 +1,171 @@
+---
+title: SQRS usage in forms
+weight: 30
+---
+
+# SQRS usage in forms
+
+## Introduction
+
+Assuming that you are already familiar with [SQRS](#) and [CRUD forms]({{< relref "CRUD-form.md" >}}) as this topic only demonstrates the usage of the SQRS approach. To apply it in your forms you need to:
+
+1. Inject a `CommandBus` or `QueryBus` instance using your class constructor.
+2. Call your command using the `CommandBus` or `QueryBus`.
+
+## Example using command bus
+
+To start with, lets inject a `CommandBus` to our _Form data handler_
+
+```yml
+#src/PrestaShopBundle/Resources/config/services/core/form/form_data_handler.yml
+
+prestashop.core.form.identifiable_object.data_handler.contact_form_data_handler:
+    class: 'PrestaShop\PrestaShop\Core\Form\IdentifiableObject\DataHandler\ContactFormDataHandler'
+    arguments:
+      - '@prestashop.core.command_bus'
+```
+
+and in `ContactFormDataHandler`
+
+```php
+namespace PrestaShop\PrestaShop\Core\Form\IdentifiableObject\DataHandler;
+
+final class ContactFormDataHandler implements FormDataHandlerInterface
+{
+    /**
+     * @var CommandBusInterface
+     */
+    private $commandBus;
+
+    /**
+     * @param CommandBusInterface $commandBus
+     */
+    public function __construct(CommandBusInterface $commandBus)
```

Comment:
include full namespace for `CommandBusInterface` at the top, so its easier for reader to find it.